content/docs/examples/multicluster/split-horizon-eds/index.md

```diff
@@ -175,24 +178,43 @@ This will be used to access pilot on `cluster1` securely using the ingress gatew
     $ kubectl get pods --context=$CTX_CLUSTER2 -n istio-system
     {{< /text >}}
 
-1. Update the gateway address in the mesh network configuration:
+1.  Determine the ingress IP and port for `cluster2`.
 
-    * Determine the gateway address for `cluster2`:
+    1.   Set the current context of `kubectl` to `CTX_CLUSTER2`
 
         {{< text bash >}}
-        $ kubectl get --context=$CTX_CLUSTER2 svc --selector=app=istio-ingressgateway -n istio-system -o jsonpath="{.items[0].status.loadBalancer.ingress[0].ip}"
-        169.61.102.93
+        $ export ORIGINAL_CONTEXT=$(kubectl config current-context)
+        $ kubectl config use-context $CTX_CLUSTER2
         {{< /text >}}
 
-    * Edit the istio config map:
+    1.   Follow the instructions in
+        [Determining the ingress IP and ports](/docs/tasks/traffic-management/ingress/#determining-the-ingress-ip-and-ports),
+        to set the `INGRESS_HOST` and `SECURE_INGRESS_PORT` environment variables.
+
+    1.  Restore the previous `kubectl` context:
+
+        {{< text bash >}}
+        $ kubectl config use-context $ORIGINAL_CONTEXT
```

Comment:
This `ORIGINAL_CONTEXT` approach seems a little ugly.

Maybe a better approach would be to get rid of the `--context` argument in all of the commands in the document and just switch back and forth between `use context $CTX_CLUSTER1` and `use context $CTX_CLUSTER2` as needed.

Reply:
@frankbu It would create too much confusion, all the switches of the context, and introduce errors if some switch is skipped. Here I just do it once, to reuse the section of determining IP and port in another task.